src/Hpack/License.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE DeriveFunctor #-}
+module Hpack.License where
+
+import qualified Distribution.License as Cabal
+import qualified Distribution.SPDX.License as SPDX
+import           Distribution.Parsec.Class (eitherParsec)
+
+data License a = DontTouch | CanSPDX a | MustSPDX a
+  deriving (Eq, Show, Functor)
+
+parseLicense :: String -> License SPDX.License
+parseLicense license = case spdxLicense of
+  Right l | isUnknown -> MustSPDX l
+  Left  _ | isUnknown -> DontTouch
+  Right l -> CanSPDX l
+  Left  _ -> case cabalLicense of
+    Right l -> CanSPDX (Cabal.licenseToSPDX l)
+    Left _ -> DontTouch
+  where
```

Comment:
I think it might be clearer to write these cases like this:

```
  Right l | isRight cabalLicense -> CanSPDX l
  Right l -> MustSPDX l
  Left  _ -> case cabalLicense of
    Right l -> CanSPDX (Cabal.licenseToSPDX l)
    Left _ -> DontTouch
```

Then we can also get rid of the hard-coded `knocnLicenses` list.

Reply:
Yes, exactly.  Right now we don't touch unknown stuff so that Cabal will fail on typos in SPDX license identifiers.  Otherwise e.g. `GPL-2.0-olny` would be converted to `LicenseRef-GPL-2.0-olny`, which I assume we don't want.

Reply:
I added one more test to make it more clear what's happening here: https://github.com/sol/hpack/pull/294/commits/4d23731ff811f0391545e3707199516a7719e5ec